### PR TITLE
fix: Update scss comment indentation

### PR DESCRIPTION
**Pull request by boyenn**
Force dedent upon hardline when working with inline scss comments.

Fixes https://github.com/prettier/prettier/issues/4878

- [X] I’ve added tests to confirm my change works. 
Used existing tests.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
*Not relevant*
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [X] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://deploy-preview-7844--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEASAtgTwLToIYAOSABABQA6UxxA5AGYQQ0kCMANMQPSfEBijlajQBGeAE7NiAJg7diAIXGUAlAG5KINiAgEYAS2gBnZKHFiIAdwAK4hMZR4ANhbyZjW4WLxgA1nBgAygTeelAA5sgwYgCucFoAFjDojgDq8XrwhsFgcAF2GXoAbhmYyOCG7iChhnBiMFZeYfjIdE41WgBWhgAe8l6+-gF46HAAMqFwLW1xIF3dAaFhjnAAitEQ8FOO7SDBYjViZWAVmrtioTApegAmMPHIABwADFoE5jUpXgRlb3AHhZMtABHdbwBo6ewgPCGbBQOBwa4I05iOAgvQohp4Jp4LY7GroPSRGIzQyLZZrDaTJCtbYzGB4YRXW73JAyEBRPB6RyLADCEHQzRQfwArKdojUACoM+w0naFWIASSgiNgATA510AEFlQEYJhlri4ABfI1AA)✨**

### CODE REVIEW

File: tests/css_less/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -3276,7 +3276,7 @@ li + li {
 
 // Usage
 @import // Comment
-  "@{themes}/tidal-wave.less";
+"@{themes}/tidal-wave.less";
```
Comment (evilebottnawi) [inline tests/css_less/__snapshots__/jsfmt.spec.js.snap]: broken

Comment (boyenn) [reply tests/css_less/__snapshots__/jsfmt.spec.js.snap]: Is expected behaviour 
```
@import "@{themes}/tidal-wave.less";
```
or the old behaviour ?

Comment (boyenn) [reply tests/css_less/__snapshots__/jsfmt.spec.js.snap]: Fixed it by checking if we're in an atRule and whether there is a prevNode. 

I'm not sure this is the right way to go about it?

File: tests/css_less/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -3276,7 +3276,7 @@ li + li {
 
 // Usage
 @import // Comment
-"@{themes}/tidal-wave.less";
+  "@{themes}/tidal-wave.less";
```
Comment (evilebottnawi) [inline tests/css_less/__snapshots__/jsfmt.spec.js.snap]: Same situation can be for `SCSS` map too

Comment (boyenn) [reply tests/css_less/__snapshots__/jsfmt.spec.js.snap]: I'm not sure how ? I tried a single value map

File: tests/css_comments/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -707,10 +707,10 @@ $my-list2:
 =====================================output=====================================
 $my-list: "foo",
   // Foo
-    "bar"; // Bar
+  "bar"; // Bar
 
 $my-list2: a // a
-  b c;
+b c;
```
Comment (evilebottnawi) [inline tests/css_comments/__snapshots__/jsfmt.spec.js.snap]: Look here

**Comment by boyenn** [comment]
Noticed it only goes wrong if the parent node is `value-paren_group`, seems like the extra identation is added there. Will workaround that instead.

**Comment by boyenn** [comment]
@evilebottnawi Not sure what to do with the failing check ?

**Comment by boyenn** [comment]
Squashed commits to trigger failing checks again.

**Comment by boyenn** [comment]
Did a rebase to keep the PR up to date.

File: src/language-css/printer-postcss.js [hunk 4]
```diff
@@ -697,6 +697,10 @@ function genericPrint(path, options, print) {
 
         // Add `hardline` after inline comment (i.e. `// comment\n foo: bar;`)
         if (isInlineValueCommentNode(iNode)) {
+          if (parentNode.type === "value-paren_group") {
```
Comment (evilebottnawi) [inline src/language-css/printer-postcss.js:700]: Potential unsafe, `value-paren_group` can be used in functions and nested function, need more tests, I just feel that it breaks something

**Comment by boyenn** [comment]
FYI: 
I'm trying to find some free time to update/add more tests. Did not abandon this yet.

**Comment by fisker** [comment]
@boyenn Friendly ping

File: src/language-css/printer-postcss.js (hunk 4 above)
Comment (boyenn) [reply src/language-css/printer-postcss.js:712]: Having a look at this now, I'm not sure what kind of tests would suffice. I'll add nested functions for now, but I'm not sure if I'm writing the kind of tests you're looking for.

**Comment by boyenn** [comment]
Having a look at this now, I'm not sure what kind of tests would suffice. I'll add nested functions for now, but I'm not sure if I'm writing the kind of tests you're looking for.

**Comment by fisker** [comment]
Thanks for fixing this!

**Pull request merged**